pkg/kubernetes/nulecule.js

```diff
@@ -172,56 +266,78 @@ define([
             if (install_dir)
                 delete_tmp();
 
-            create = create_tmp()
-                .done(function(dir){
-                    install_dir = dir.trim();
-                    debug("created directory :"+ install_dir);
-                    runner = installrun("install", image)
-                        .progress(function(msg) {
-                            deferred.notify(msg);
-                        })
-                        .done(function() {
-                            debug("reading " + install_dir + "/answers.conf.sample");
-                            var file = cockpit.file(install_dir + "/answers.conf.sample");
-                            file.read()
-                                .done(function(content){
-                                    if(content) {
-                                        var jdata = JSON.parse(content);
-                                        debug(jdata);
-                                        self.answers = jdata;
-                                        deferred.resolve(jdata);
-                                    } else {
-                                        var message = _("Unable to read answer.conf.sample file.");
+            check_atomic_version()
+                .done(function() {
```

Comment:
This entire block of code (the massive indentation) has become way way too unreadable and hard to follow. Please break it out into a few functions. If you need help, let me know and I can give more advice.